File: 7-reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        
        num=abs(x)
        rev=0
        
        while num!=0:
            digit=num%10

            if rev>(2**31 -1)//10:
                return 0
            rev = rev*10+digit
            num=num//10
        
        if x<0:
            return -rev
        else:
            return rev
            

File: 7-reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
        int p = Math.abs(x);
        int r=0;
        while(p>0){
            int digit = p%10;
            if(r>(Integer.MAX_VALUE-digit)/10){
                return 0;
            }
            r = r*10+digit;
            p/=10;
        }
        if(x<0){
            return -1*r;
        }
        else{
            return r;
        }
    }
}